### DISCUSSION TITLE
Total novice trying to deploy inventree on synology NAS with docker

### DISCUSSION THREAD

**Discussion by Downhunter** [inventree/InvenTree#3955] [Q&A]
Hi!

I am a total docker and Inventree novice.
I am trying to deploy Inventree on synology nas, and i am having a bit of trouble.
Im trying to follow the docker documentation, but im just getting "Error: Missing required database configuration value ENGINE".
Its probably because i havent set up a database, and dont really know how to do that. My quastion is, does anyone have a synology docker deployment guide.
Im running a DS1817+ and a ds920+, and have been trying to set it up on my personal DS1817+ to have my own at home. And when i know how that works, i would deploy a new one on the DS920+ that we have at the scouting group im a part of.

Any help is much appreciated.

**Comment by wolflu05** [comment]
Hello, do you followed [this](https://docs.inventree.org/en/latest/start/docker_prod/#initial-database-setup) guide and make sure that you did the linked step? This step is very important to set up the database before starting the inventree server.

**Comment by matmair** [comment]
Synology docker has a rather interesting admin interface. Please read the docker guides [we provide](https://docs.inventree.org/en/latest/start/docker/) fully, every step matters. If you run into further problems just ask here.
I tried packaging natively for DSM 7 a while ago and had to surrender. So no native package right now.

**Reply by Downhunter** [reply]
Okay so i have now gotten a little bit futher... i think..
I have through SSH gotten the the 4 container up and running with docker compose.
I used "cd /volume1/docker/Inventree_test" to get into the folder where i got my ".env", "nginx.prod.conf" and "docker-compose.yml"
And then i ran "sudo docker-compose up --detach" to get it to compose. And it ran fine, got all 4 containers up and running and everything.
But the "inventree-db" keeps crashing and rebooting, and the logs only say "mkdir: cannot create directory ‘/var/lib/postgresql/data’: Permission denied".
I tried mounting it through volume settings in the synology docker ui, but when i try mounting "/var/lib/postgresql/data" to a folder it just says "Unable to perform this operation, possibly because the network connection is unstable or the system is busy. Please try again later."
When i just try to mount "/var/lib/postgresql" to the same folder it can do it, but when i launch the container it just crashes, but it has created a "data" folder but stil the samme "mkdir: cannot create directory ‘/var/lib/postgresql/data’: Permission denied" error code in the log.
I have also done it all while giving the container "execute container using high privlege" without any change.
Am i just stupid or is there somthing wrong?

**Reply by miggland** [reply]
`/var/lib/postgresql/data`  sounds like the path inside the container.

Are you using the correct path in your main system?

**Reply by Downhunter** [reply]
No idea.. 🫤
When i run the compose file it is in "/volume1/docker/inventree_test"
It starts up fine but there is just this permissions error and then it just reboots

**Reply by miggland** [reply]
Can you post the docker-compose and .env files?

Docker mounts volumes from the filesystem to a certain location in the container. These paths are different, and it's important to use the right one in the right situation. Probably it's a good idea to read up on this in Docker docs / some tutorial if you're unclear about it.

Elevated privileges shouldn't be necessary, however, Synology always has been a bit special. Don't have access to one any longer, and never ran Docker on the ones I had.

**Reply by Downhunter** [reply]
Yeah i will post my docker-compose and .env file as soon as i get home from work 😁

**Reply by Downhunter** [reply]
The docker-compose.yml
[docker-compose.yml.txt](https://github.com/inventree/InvenTree/files/10511158/docker-compose.yml.txt)


And the .env file (all values are for testing purposes and will be different in an actual deployment)
[.env.txt](https://github.com/inventree/InvenTree/files/10511164/default.env.txt)

And they are not .txt files when deployed

**Reply by miggland** [reply]
Alright, the directory you're looking for on the Synology is this:
`/volume1/docker/Inventree_test`

This should exist, and be writable (for the user running Docker, `root`?)

This directory will be mounted into all the containers, except `inventree-cache`. Which sub-directory each container writes to varies.

Postgres will create a directory in there called `pgdb`.

**Reply by Downhunter** [reply]
Hi sorry for the late response, ive been trying some thing without any luck.

These are the containers the i go
![All Docker containers](https://user-images.githubusercontent.com/118010506/216379649-bbbbe847-2f68-4fdb-94da-fa9f9051c291.png)

And when i try to start Inventree-db i get this error in the log
![Docker inventree-db log](https://user-images.githubusercontent.com/118010506/216380114-bf6d4cd3-61f9-408c-8738-d866885ff651.png)

When i try to mount the path like this
![Docker inventree-db path](https://user-images.githubusercontent.com/118010506/216380409-62fefae4-8377-4ca9-bf44-19127438db3b.png)

I get this error
![Docker inventree-db path error](https://user-images.githubusercontent.com/118010506/216380490-1b2b5764-375a-4731-bf14-c82c5ab260b9.png)

But i can mount "/var/lib/postgresql". But then when i try to run the container again it just crashes again and the logs say this
![Docker inventree-db log 2](https://user-images.githubusercontent.com/118010506/216381123-dbd7ca9e-a24c-47ba-a7a5-b0fd04d824eb.png)

But it has created two folders, a "data" folder and a "pgdb" folder in "/volume1/docker/Inventree_test"
![Docker Inventree folders](https://user-images.githubusercontent.com/118010506/216381560-5ad4ae90-cfbe-433e-b3e0-1026be817c6c.png)

How can it be that it can crate folders, but it dosent have premissions to write in them??
Thanks for the help in advance.

**Reply by miggland** [reply]
Are you trying to start only the DB container, or all of them? I would suggest trying just the DB container. Maybe try with a minimal docker-compose for only a postgres image.

Can you log in via SSH and output the ownership and permissions of docker/Inventree_test and its contents?

Something like:
```
ls -al docker/
ls -al docker/Inventree_test
```

**Reply by Downhunter** [reply]
When i try the comands it just says this.

![Skærmbillede 2023-02-13 163034](https://user-images.githubusercontent.com/118010506/218500796-c689c7c1-ab9c-4133-a644-1ac8df7f44c8.jpg)

**Reply by miggland** [reply]
What I meant was "something like", in your case would have been:
```
ls -al /volume1/docker
ls -al /volume1/docker/Inventree_test
```
alternatively switching to /volume1/ with `cd` before the commands.

Anyway, I see a potential problem: your pgdb folder is owned by user 999 and root, and is not writeable by any other user.

Try this:
Find your user and group ID (see: https://www.synoforum.com/resources/how-to-find-uid-userid-and-gid-groupid.77/)

Change ownership of the pgdb folder:
`chown -R miggestok:users /volume1/docker/Inventree_test/pgdb`

Change you `docker-compose.yml` to add a line underneath "image: postgres:13":
`user: "1031:100"`

Use your user ID, and group ID.

Start the PG container.

**Reply by Downhunter** [reply]
This is weird... I can take ownership of the folder through DSM but as soon as i launch the container it changes it back to 999.
![Skærmbillede 2023-02-13 183103](https://user-images.githubusercontent.com/118010506/218530468-f267cc7b-961f-4039-9787-68724705d41e.jpg)

Even if i delete the "pgdb" folder and create a new one witch i then own it stil changes it back to 999
![Skærmbillede 2023-02-13 183630](https://user-images.githubusercontent.com/118010506/218531055-8ed3a109-c0ad-455e-8553-cbdd12086c94.jpg)

**Comment by wintedltd** [comment]
Hi dude, did you finish deploying inventree on synology NAS with docker?

**Comment by miggland** [comment]
Yes, that's why you need the extra line in your docker file. Otherwise, the user in the docker container will change the permissions on start.

**Reply by Downhunter** [reply]
That dosent seem to change anything?
You mean like this right?
[docker-compose.yml.txt](https://github.com/inventree/InvenTree/files/10725233/docker-compose.yml.txt)

**Comment by miggland** [comment]
Yes, that looks like what I had in mind.. don't really know what's going on.

Have you tried searching for similar issues regarding PG and Synology Docker? Seems like this shouldn't have anything to do with InvenTree itself.

**Reply by Downhunter** [reply]
Yeah i have tried a little bit, but i couldn't really find any thing but i will keep trying.
Thanks for the help tho! 😁
